doc/release/release_2.5.rst

```diff
@@ -0,0 +1,450 @@
+- add Pathlib support to work with files.
+- improve performance.
+- Updated docs and tests.
+- Removed code designed to work with Python 2.
+
+New Functions:
+
+- lukes_partitioning
+- triadic analysis functions
+- functions for trophic levels analysis
+- d_separated
+- is_regular and other regular graph measures
+- graph_hash using Weisfeiler Lehman methods
+- common_neighbor_centrality (CCPA link prediction)
+- max_weight_clique
+- path_weight and is_path
+- rescale_layout_dict
+
+New generators:
+
+- paley_graph
+- interval_graph
+
+New layouts:
+
+- multipartite_layout
+
+
+Improvements
+------------
+
+- Add governance documents, developer guide and community structures
+- Implement explicit deprecation policy.
+- Initiate an NX Enhancement Proposal (NXEP) system
+- optimize single_source_shortest_path
+- improved consistent "weight" specification in shortest_path routines
+- improved line color
+- better search engine treatment of docs
+- lattice and grid_graph and grid_2d_graph can use dim=tuple
+- fix initializer of kamada_kawai_layout algorithm
+- moral and threshold functions now included in namespace and docs
+- scale arrows better when drawing
+- more uniform creation of random lobster graphs
+- allow editing graph during iteration over connected_components
+- better column handling in coversion of pandas DataFrame
+- allow simrank_similarity with directed graph input
+- ensure VoteRank ability is nonnegative
+- speedup kernighan_lin_bisection
+- speedup negative weight cycle detection
+- tree_isomorphism
+- rooted_tree_isomorphism
+
+API Changes
+-----------
+
+- enabled "copy" flag parameter in `contracted_nodes`
+- allow partially periodic lattices
+- return value for minimum_st_node_cut now always a set
+- removed unused "has_numpy" argument from create_py_random_state
+- fixed return values when drawing empty nodes and edges
+- allow sets and frozensets of edges as input to nx.Graph()
+- "weight" can be function for astar, directional_dijksta, all_shortest_path
+- allow named key ids for GraphML edge writing
+- all keywords are now checked for validity in nx.draw and friends
+- EdgeDataView "in" operator checks if nodes are "in nbunch"
+- remove completeness condition from minimum weight full matching
+- option to sort neighbors in bfs traversal
+- draw_networkx accepts numpy array for edgelist
+- relabel_nodes with 2 nodes mapped to same node can now create multiedge
+- steiner_tree works with MultiGraph
+
+Deprecations
+------------
+
+- [`#3680 <https://github.com/networkx/networkx/pull/3680>`_]
+  Deprecate `make_str(x)` for `str(x)`.
+  Deprecate `is_string_like(obj)` for `isinstance(obj, str)`.
+
+- [`#3725 <https://github.com/networkx/networkx/pull/3725>`_]
+  Deprecate `literal_stringizer` and `literal_destringizer`.
+
+- [`#3983 <https://github.com/networkx/networkx/pull/3983>`_]
+  Deprecate `reversed` context manager in favor of `reviersed_view`
+
+- [`#4045 <https://github.com/networkx/networkx/pull/4045>`_]
```

Comment:
I am not sure this should be listed under deprecations.  Perhaps, it would be sufficient to just add
```
- Reduce numpy.matrix usage which is deprecated.
```
under ``Improvements``.

@rossbar Thoughts?

Reply:
Agreed, there are no actual deprecations/future warnings related to this (yet). I think it should be listed with the normal changes that aren't related to deprecations.